### PR TITLE
Layer list619

### PR DESCRIPTION
**Pull request by wnordmann**
Similar to earlier layer-list pr, but this time with less commits.

### CODE REVIEW

**Comment by wnordmann** [comment]
https://github.com/boundlessgeo/sdk/pull/583

File: examples/layer-list/app.jsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  };
+
+  // add 200 random points to the map on startup
+  addRandomPoints(200);
+
+  // add the wms source
+  store.dispatch(mapActions.addSource('states', {
+    type: 'raster',
+    tileSize: 256,
+    tiles: ['https://ahocevar.com/geoserver/gwc/service/wms?SERVICE=WMS&VERSION=1.1.1&REQUEST=GetMap&FORMAT=image/png&SRS=EPSG:900913&LAYERS=topp:states&STYLES=&WIDTH=256&HEIGHT=256&BBOX={bbox-epsg-3857}'],
```
Comment (bartvde) [inline examples/layer-list/app.jsx]: seems Andreas's WMS is down, I think we should switch back to demo.boundlessgeo.com here

Comment (bartvde) [inline examples/layer-list/app.jsx]: ah it's just came back up, but given that boundlessgeo demo has been redone, we can use that again I think anyway

File: examples/layer-list/app.jsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
+/** Demo of layer list in an SDK map.
+ *
+ *  Contains a Map and demonstrates adding many types of layers
+ *  And a layer list componet to manage the layers
```
Comment (bartvde) [inline examples/layer-list/app.jsx]: component

File: examples/layer-list/app.jsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+      // addFeatures with the features, source name, and crs
+      .then(json => store.dispatch(mapActions.addFeatures(sourceName, json)));
+  };
+
+  // This is called by the onClick, keeping the onClick HTML clean
+  const runFetchGeoJSON = () => {
+    const url = './data/airports.json';
+    addLayerFromGeoJSON(url, 'dynamic-source');
+  };
+  runFetchGeoJSON();
```
Comment (bartvde) [inline examples/layer-list/app.jsx]: we can simply use a source with a url as data now

```store.dispatch(mapActions.addSource('dynamic-source', { type: 'geojson', data: './data/airports.json' }));```

File: examples/layer-list/components/layerList.jsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+import * as mapActions from '@boundlessgeo/sdk/actions/map';
+
+const isLayerVisible = (layer) => {
+  if (typeof layer.layout !== 'undefined') {
+    return layer.layout.visibility !== 'none';
+  }
+  return true;
+};
+
+class layerList extends React.PureComponent {
```
Comment (bartvde) [inline examples/layer-list/components/layerList.jsx]: Personally I prefer LayerList as the component name (capital L)

File: examples/layer-list/components/layerList.jsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  }
+  buildListOfLayers(layers) {
+    const list = [];
+    for (let i = layers.length - 1, ii = 0; i >= ii; i--) {
+      const layer = layers[i];
+      const is_checked = isLayerVisible(layer);
+
+      const checkbox = (<input
+        type="checkbox"
+        onChange={() => { }}
```
Comment (bartvde) [inline examples/layer-list/components/layerList.jsx]: is the empty onChange needed?

File: examples/layer-list/components/layerList.jsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    id: PropTypes.string,
+    source: PropTypes.string,
+  })).isRequired,
+};
+
+layerList.defaultProps = {
+
+};
+
+// export default layerList;
```
Comment (bartvde) [inline examples/layer-list/components/layerList.jsx]: remove comment

File: examples/layer-list/index.html [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+<head>
+    <title>Layer List Example</title>
+    <link rel="stylesheet" type="text/css" href="/dist/sdk.css"/>
+    <link rel="stylesheet" type="text/css" href="../examples.css"/>
+    <link rel="stylesheet" type="text/css" href="app.css"/>
+</head>
+<body>
+
+    <div id="header">
+        Drawing Example
```
Comment (bartvde) [inline examples/layer-list/index.html]: incorrect header

File: examples/layer-list/index.html [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    </div>
+
+    <div id="controls">
+    </div>
+
+    <div class="info">
+        <h3>Features in this demo</h3>
+        <ul class="info-list">
+            <li>Add Random Points</li>
+            <li>Add WMS States layer</li>
```
Comment (bartvde) [inline examples/layer-list/index.html]: list not complete

**Comment by bartvde** [comment]
@wnordmann are you able to change this PR to be on master?

**Pull request closed without merging**